scripts/pre_download_models.py

```
@@ -0,0 +1,66 @@
+        raise
+
+    print('🚀 Pre-downloading Whisper model...')
+    try:
+        # Check numpy availability first
+        try:
+            import numpy
+            print(f'✅ Numpy {numpy.__version__} available')
+        except ImportError:
+            print('⚠️ Installing numpy...')
+            import subprocess
+            subprocess.check_call([sys.executable, '-m', 'pip', 'install', 'numpy'])
```

Ruff (0.12.2)

`subprocess` call: check for execution of untrusted input

(S603)